job-manager-it/src/test/java/uk/gov/moj/cpp/jobmanager/it/JobServiceIT.java

```diff
@@ -0,0 +1,288 @@
+            jobService.getUnassignedJobsFor(workerId);
+
+            collectInfo(workerId);
+
+        }
+
+        userTransaction.commit();
+
+        detectDuplicates();
+
+        assertThat(testJobJdbcRepository.jobsProcessed(), CoreMatchers.is(30));
+
+    }
+
+    @Test
+    public void shouldUpdateJobsThatHaveAWorkerIdAndBeenIdleForMoreThanOneHourAndReadyToStart() throws Exception {
+
+        userTransaction.begin();
+        testJobJdbcRepository.cleanJobTables();
+        testJobJdbcRepository.createJobs(20);
+        testJobJdbcRepository.createIdleJobs(5, of(now().minus(65, MINUTES)),  now().plus(30, MINUTES));
+        testJobJdbcRepository.createIdleJobs(5, of(now().minus(65, MINUTES)),  now());
+        testJobJdbcRepository.createIdleJobs(10, of(now().minus(30, MINUTES)), now());
+        userTransaction.commit();
+
+
+        userTransaction.begin();
+        for (int i = 0; i < 4; i++) {
+
+            final UUID workerId = randomUUID();
+            jobService.getUnassignedJobsFor(workerId);
+
+            collectInfo(workerId);
+
+        }
+
+        userTransaction.commit();
+        detectDuplicates();
+        assertThat(testJobJdbcRepository.jobsProcessed(), CoreMatchers.is(25));
+
+
+    }
+
+    @SuppressWarnings("squid:S2925")
+    @Test
+    public void shouldNotPerformDuplicateJobUpdates() throws Exception {
+
+        userTransaction.begin();
+        testJobJdbcRepository.cleanJobTables();
+        testJobJdbcRepository.createJobs(1000);
+        userTransaction.commit();
+
+        final CyclicBarrier gate = new CyclicBarrier(21);
+
+        IntStream.range(0, 20)
+                 .mapToObj(threadNo -> getThreadCurrentImpl(gate, threadNo))
+                 .forEach(t -> t.start());
+
+        gate.await();
+
+        Thread.sleep(3000);
+
+        detectDuplicates();
+
+        assertThat(testJobJdbcRepository.jobsProcessed(), CoreMatchers.is(200));
+
+    }
+
+
+    private Thread getThreadCurrentImpl(final CyclicBarrier gate, final int count) {
+        return new Thread(() -> {
+            try {
+                gate.await();
+
+                String workerThreadName = new StringBuilder("T").append(count).append(" Worker: ").toString();
+                final UUID workerId = randomUUID();
+                System.out.println(workerThreadName);
+
+                userTransaction.begin();
+
+                Stream<Job> jobStream = jobService.getUnassignedJobsFor(workerId);
+
+                System.out.println(workerId + " jobs selected: " + jobStream.count());
+
+                collectInfo(workerId);
+
+                try {
+                    userTransaction.commit();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                try {
+                    userTransaction.rollback();
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
remove printStackTrace() and don't catch the exceptions the the test method throw Exception instead